clang/lib/AST/ByteCode/Compiler.cpp

```diff
@@ -1222,6 +1224,117 @@ bool Compiler<Emitter>::VisitComplexBinOp(const BinaryOperator *E) {
   return true;
 }
 
+template <class Emitter>
+bool Compiler<Emitter>::VisitVectorBinOp(const BinaryOperator *E) {
+  assert(E->getType()->isVectorType());
```

Comment:
Add an assertion that both LHS and RHS are also of vector type.

Reply:
Done.